src/transformers/models/omnivore/modeling_omnivore.py

```diff
@@ -0,0 +1,987 @@
+    ImageClassifierOutputWithNoAttention,
+)
+from ...modeling_utils import PreTrainedModel
+from ...utils import add_code_sample_docstrings, add_start_docstrings, add_start_docstrings_to_model_forward, logging
+from .configuration_omnivore import OmnivoreConfig
+
+
+logger = logging.get_logger(__name__)
+
+# General docstring
+_CONFIG_FOR_DOC = "OmnivoreConfig"
+_FEAT_EXTRACTOR_FOR_DOC = "OmnivoreFeatureExtractor"
+
+# Base docstring
+_CHECKPOINT_FOR_DOC = "anugunj/omnivore-swinT"
+_EXPECTED_OUTPUT_SHAPE = [1, 768, 1, 7, 7]
+
+# Image classification docstring
+_IMAGE_CLASS_CHECKPOINT = "anugunj/omnivore-swinT"
+_IMAGE_CLASS_EXPECTED_OUTPUT = "tabby, tabby cat"
+
+OMNIVORE_PRETRAINED_MODEL_ARCHIVE_LIST = [
+    "anugunj/omnivore-swinT",
+    # See all Omnivore models at https://huggingface.co/models?filter=omnivore
+]
+
+
+def drop_path(input, drop_prob=0.0, training=False, scale_by_keep=True):
+    """
+    Drop paths (Stochastic Depth) per sample (when applied in main path of residual blocks).
+    """
+    if drop_prob == 0.0 or not training:
+        return input
+    keep_prob = 1 - drop_prob
+    shape = (input.shape[0],) + (1,) * (input.ndim - 1)  # work with diff dim tensors, not just 2D ConvNets
+    random_tensor = input.new_empty(shape).bernoulli_(keep_prob)
+    if keep_prob > 0.0 and scale_by_keep:
+        random_tensor.div_(keep_prob)
+    return input * random_tensor
+
+
+class OmnivoreDropPath(nn.Module):
+    def __init__(self, drop_prob=None):
+        super().__init__()
+        self.drop_prob = drop_prob
+
+    def forward(self, x: torch.Tensor):
+        return drop_path(x, self.drop_prob, self.training)
+
+
+class OmnivoreIm2Video(nn.Module):
+    """Convert Image into a trivial video"""
+
+    def forward(self, pixel_values):
+        if pixel_values.ndim == 4:
+            return pixel_values.unsqueeze(2)
+        elif pixel_values.ndim == 5:
+            return pixel_values
+        else:
+            raise ValueError(f"Dimension incorrect {pixel_values.shape}")
+
+
+class OmnivoreSwinMLPLayer(nn.Module):
+    def __init__(self, in_features, hidden_features=None, out_features=None, dropout_rate=0.0, act_layer=nn.GELU):
+        super().__init__()
+        out_features = out_features or in_features
+        hidden_features = hidden_features or in_features
+        self.linear1 = nn.Linear(in_features, hidden_features)
+        self.activation = act_layer()
+        self.linear2 = nn.Linear(hidden_features, out_features)
+        self.drop_out = nn.Dropout(dropout_rate)
+
+    def forward(self, hidden_state):
+        hidden_state = self.linear1(hidden_state)
+        hidden_state = self.activation(hidden_state)
+        hidden_state = self.drop_out(hidden_state)
+        hidden_state = self.linear2(hidden_state)
+        hidden_state = self.drop_out(hidden_state)
+        return hidden_state
+
+
+def window_partition(input_feature, window_size):
+    batch_size, D, height, width, channels = input_feature.shape
```

Comment:
```suggestion
    batch_size, D, height, width, channels = input_feature.shape
```
What's D here? You can replace it by something more understandable, like time.

Also wonder why channels are last here

Reply:
Interestingly, the official implementation of Omnivore uses this shape: `B x C x T x H x W` ([reference](https://github.com/facebookresearch/omnivore/blob/main/omnivore/models/omnivore_model.py#L58)).